Python error when trying to make an OpenEmu recipe

I noticed that there isn't an OpenEmu.munki recipe yet, so I wanted to make one.  @sheagcraig already made .download and .jss recipes, so I was hoping to use Recipe Robot to ignore his recipes, then alter the newly-created .munki recipe to use his .download recipe as the parent.

I couldn't figure out how to "ignore existing" from the GUI (though I'm pretty sure you told me at PSU 2016), so I turned to using the script.  Here's what I get when inputting the application:

```
mike$ "/Applications/Recipe Robot.app/Contents/Resources/scripts/recipe-robot" /Applications/OpenEmu.app --ignore-existing --verbose
[WARNING] Will build recipes even if they already exist in "autopkg search" results. Please don't upload duplicate recipes to public AutoPkg repos on GitHub.

                      -----------------------------------
                     |  Welcome to Recipe Robot v1.0.4.  |
                      -----------------------------------
                                \   _[]_
                                 \  [oo]
                                   d-||-b
                                     ||
                                   _/  \_

Processing /Applications/OpenEmu.app ...
Input path looks like an app.
Validating app...
    App seems valid
Getting app name...
    App name is: OpenEmu
Getting bundle identifier...
    Bundle identifier is: org.openemu.OpenEmu
Checking for a Sparkle feed...
    Sparkle feed is: https://raw.github.com/OpenEmu/OpenEmu-Update/master/appcast.xml
Getting information from Sparkle feed...
    The Sparkle feed provides a version number
    The latest version is 2.0.3
    Download URL is: https://github.com/OpenEmu/OpenEmu/releases/download/v2.0.3/OpenEmu_2.0.3.zip
Getting GitHub repo...
    GitHub repo is: OpenEmu/OpenEmu
Getting GitHub description...
    GitHub description is: :video_game: Retro video game emulation for macOS
Getting information from latest GitHub release...
    GitHub release download format is: zip
    GitHub release download URL is: https://github.com/OpenEmu/OpenEmu/releases/download/v2.0.3/OpenEmu_2.0.3.zip
    Download URL is: https://github.com/OpenEmu/OpenEmu/releases/download/v2.0.3/OpenEmu_2.0.3.zip
Downloading file for further inspection...
    Downloaded to /Users/mike/Library/Caches/Recipe Robot/2016-10-25_22-34-22_957197/OpenEmu_2.0.3.zip
Determining download format...
    File extension is zip
Getting developer name from GitHub...
    GitHub developer is: None
Downloading file for further inspection...
    Downloaded to /Users/mike/Library/Caches/Recipe Robot/2016-10-25_22-34-22_957197/OpenEmu_2.0.3.zip
Determining download format...
    File extension is zip
Looking for version key...
    Version key is: CFBundleShortVersionString (2.0.3)
Looking for app icon...
    App icon is: /Applications/OpenEmu.app/Contents/Resources/OpenEmu
Gathering code signature information...
    App is not signed
[ERROR] Recipe Robot exploded with unexpected error: 'NoneType' object has no attribute 'replace'
Traceback (most recent call last):
  File "/Applications/Recipe Robot.app/Contents/Resources/scripts/recipe-robot", line 98, in main
    time, _ = recipe_robot_lib.generate_recipes(facts, prefs)  # pylint: disable=assignment-from-no-return
  File "/Applications/Recipe Robot.app/Contents/Resources/scripts/recipe_robot_lib/tools.py", line 117, in run_func
    result = func(*args, **kwargs)
  File "/Applications/Recipe Robot.app/Contents/Resources/scripts/recipe_robot_lib/recipe_generator.py", line 95, in generate_recipes
    facts["developer"].replace("/", "-"))
AttributeError: 'NoneType' object has no attribute 'replace'

[ERROR] 
                                    _[]_
                                    [xx]
                                   q-||-p
                                     ||
                                   _/  \_
```

The GUI application gets to nearly the same point, and stops at the duplicate recipe check (instead of erroring):

```

                      -----------------------------------
                     |  Welcome to Recipe Robot v1.0.4.  |
                      -----------------------------------
                                \   _[]_
                                 \  [oo]
                                   d-||-b
                                     ||
                                   _/  \_

Processing /Applications/OpenEmu.app ...
Input path looks like an app.
Validating app...
    App seems valid
Getting app name...
    App name is: OpenEmu
Getting bundle identifier...
    Bundle identifier is: org.openemu.OpenEmu
Checking for a Sparkle feed...
    Sparkle feed is: https://raw.github.com/OpenEmu/OpenEmu-Update/master/appcast.xml
Getting information from Sparkle feed...
    The Sparkle feed provides a version number
    The latest version is 2.0.3
    Download URL is: https://github.com/OpenEmu/OpenEmu/releases/download/v2.0.3/OpenEmu_2.0.3.zip
Getting GitHub repo...
    GitHub repo is: OpenEmu/OpenEmu
Getting GitHub description...
    GitHub description is: :video_game: Retro video game emulation for macOS
Getting information from latest GitHub release...
    GitHub release download format is: zip
    GitHub release download URL is: https://github.com/OpenEmu/OpenEmu/releases/download/v2.0.3/OpenEmu_2.0.3.zip
    Download URL is: https://github.com/OpenEmu/OpenEmu/releases/download/v2.0.3/OpenEmu_2.0.3.zip
Downloading file for further inspection...
    10.00%
    19.99%
    29.99%
    39.99%
    49.98%
    59.98%
    69.98%
    79.97%
    89.97%
    99.97%
    Downloaded to /Users/mike/Library/Caches/Recipe Robot/2016-10-26_15-31-32_358272/OpenEmu_2.0.3.zip
Determining download format...
    File extension is zip
Getting developer name from GitHub...
    GitHub developer is: None
Downloading file for further inspection...
    10.00%
    19.99%
    29.99%
    39.99%
    49.98%
    59.98%
    69.98%
    79.97%
    89.97%
    99.97%
    Downloaded to /Users/mike/Library/Caches/Recipe Robot/2016-10-26_15-31-32_358272/OpenEmu_2.0.3.zip
Determining download format...
    File extension is zip
Looking for version key...
    Version key is: CFBundleShortVersionString (2.0.3)
Looking for app icon...
    App icon is: /Applications/OpenEmu.app/Contents/Resources/OpenEmu
Gathering code signature information...
    App is not signed
Searching for existing AutoPkg recipes for OpenEmu...
    Found existing recipe(s):
        OpenEmu.jss.recipe
        OpenEmu.download.recipe
        OpenEmu.pkg.recipe
[ERROR] Sorry, AutoPkg recipes already exist for this app, and I can't blend new recipes with existing recipes.

Here are my suggestions:
    - See if one of the above recipes meets your needs, either as-is or using an override.
    - Write your own recipe using one of the above as the ParentRecipe.
    - Use Recipe Robot to assist in the creation of a new child recipe, as seen here:
      https://youtu.be/5VKDzY8bBxI?t=2829
[ERROR] 
                                    _[]_
                                    [xx]
                                   q-||-p
                                     ||
                                   _/  \_
```

Any idea what's wrong?

--- a/scripts/recipe_robot_lib/inspect.py
+++ b/scripts/recipe_robot_lib/inspect.py
@@ -313,7 +313,7 @@ def inspect_app(input_path, args, facts):
         else:
             if "CFBundleVersion" in info_plist:
                 version_key = "CFBundleVersion"
-        if version_key != "":
+        if version_key not in ("", None):
             robo_print("Version key is: %s (%s)" %
                        (version_key, info_plist[version_key]), LogLevel.VERBOSE, 4)
             facts["version_key"] = version_key
@@ -329,7 +329,7 @@ def inspect_app(input_path, args, facts):
             icon_path = os.path.join(input_path, "Contents", "Resources", info_plist["CFBundleIconFile"])
         else:
             facts["warnings"].append("Can't determine app icon.")
-        if icon_path != "":
+        if icon_path not in ("", None):
             robo_print("App icon is: %s" % icon_path, LogLevel.VERBOSE, 4)
             facts["icon_path"] = icon_path
 
@@ -391,7 +391,7 @@ def inspect_app(input_path, args, facts):
             facts["codesign_reqs"] = codesign_reqs
             robo_print("%s authority names recorded" % len(codesign_authorities), LogLevel.VERBOSE, 4)
             facts["codesign_authorities"] = codesign_authorities
-        if developer != "":
+        if developer not in ("", None):
             robo_print("Developer: %s" % developer, LogLevel.VERBOSE, 4)
             facts["developer"] = developer
 
@@ -569,7 +569,7 @@ def inspect_bitbucket_url(input_path, args, facts):
     r_obj = re.search(r"(?<=https://bitbucket\.org/)[\w-]+/[\w-]+", input_path)
     if r_obj is not None:
         bitbucket_repo = r_obj.group(0)
-    if bitbucket_repo != "":
+    if bitbucket_repo not in ("", None):
         robo_print("BitBucket repo is: %s" % bitbucket_repo, LogLevel.VERBOSE, 4)
         facts["bitbucket_repo"] = bitbucket_repo
 
@@ -617,9 +617,9 @@ def inspect_bitbucket_url(input_path, args, facts):
         if "app_name" not in facts:
             app_name = ""
             robo_print("Getting app name...", LogLevel.VERBOSE)
-            if parsed_repo.get("name", "") != "":
+            if parsed_repo.get("name", "") not in ("", None):
                 app_name = parsed_repo["name"]
-            if app_name != "":
+            if app_name not in ("", None):
                 robo_print("App name is: %s" % app_name, LogLevel.VERBOSE, 4)
                 facts["app_name"] = app_name
 
@@ -627,7 +627,7 @@ def inspect_bitbucket_url(input_path, args, facts):
         developer = ""
         if "developer" not in facts:
             developer = parsed_repo["owner"]["display_name"]
-        if developer != "":
+        if developer not in ("", None):
             robo_print("BitBucket owner full name "
                        "is: %s" % developer, LogLevel.VERBOSE, 4)
             facts["developer"] = developer
@@ -636,9 +636,9 @@ def inspect_bitbucket_url(input_path, args, facts):
         if "description" not in facts:
             description = ""
             robo_print("Getting BitBucket description...", LogLevel.VERBOSE)
-            if parsed_repo.get("description", "") != "":
+            if parsed_repo.get("description", "") not in ("", None):
                 description = parsed_repo["description"]
-            if description != "":
+            if description not in ("", None):
                 robo_print("BitBucket description is: %s" % description, LogLevel.VERBOSE, 4)
                 facts["description"] = description
             else:
@@ -657,14 +657,14 @@ def inspect_bitbucket_url(input_path, args, facts):
                             download_format = this_format
                             download_url = asset["links"]["self"]["href"]
                             break
-            if download_format != "":
+            if download_format not in ("", None):
                 robo_print("BitBucket release download format "
                            "is: %s" % download_format, LogLevel.VERBOSE, 4)
                 facts["download_format"] = download_format
             else:
                 facts["warnings"].append(
                     "Could not detect BitBucket release download format.")
-            if download_url != "":
+            if download_url not in ("", None):
                 robo_print("BitBucket release download URL "
                            "is: %s" % download_url, LogLevel.VERBOSE, 4)
                 facts["download_url"] = download_url
@@ -1088,7 +1088,7 @@ def inspect_github_url(input_path, args, facts):
         github_repo = parsed_url.netloc.split(".")[0] + "/" + path[0]
     else:
         github_repo = path[0] + "/" + path[1]
-    if github_repo != "":
+    if github_repo not in ("", None):
         robo_print("GitHub repo is: %s" % github_repo, LogLevel.VERBOSE, 4)
         facts["github_repo"] = github_repo
 
@@ -1161,7 +1161,7 @@ def inspect_github_url(input_path, args, facts):
             robo_print("Getting app name...", LogLevel.VERBOSE)
             if parsed_repo.get("name", None) is not None:
                 app_name = parsed_repo["name"]
-            if app_name != "":
+            if app_name not in ("", None):
                 robo_print("App name is: %s" % app_name, LogLevel.VERBOSE, 4)
                 facts["app_name"] = app_name
 
@@ -1171,7 +1171,7 @@ def inspect_github_url(input_path, args, facts):
             robo_print("Getting GitHub description...", LogLevel.VERBOSE)
             if parsed_repo.get("description", None) is not None:
                 description = parsed_repo["description"]
-            if description != "":
+            if description not in ("", None):
                 robo_print("GitHub description is: %s" % description,
                            LogLevel.VERBOSE, 4)
                 facts["description"] = description
@@ -1191,7 +1191,7 @@ def inspect_github_url(input_path, args, facts):
                             download_format = this_format
                             download_url = asset["browser_download_url"]
                             break
-            if download_format != "":
+            if download_format not in ("", None):
                 robo_print("GitHub release download format "
                            "is: %s" % download_format,
                            LogLevel.VERBOSE, 4)
@@ -1199,7 +1199,7 @@ def inspect_github_url(input_path, args, facts):
             else:
                 facts["warnings"].append(
                     "Could not detect GitHub release download format.")
-            if download_url != "":
+            if download_url not in ("", None):
                 robo_print("GitHub release download URL "
                            "is: %s" % download_url, LogLevel.VERBOSE, 4)
                 facts["download_url"] = download_url
@@ -1215,7 +1215,7 @@ def inspect_github_url(input_path, args, facts):
                        LogLevel.VERBOSE)
             if "name" in parsed_user:
                 developer = parsed_user["name"]
-            if developer != "":
+            if developer not in ("", None):
                 robo_print("GitHub developer "
                            "is: %s" % developer, LogLevel.VERBOSE, 4)
                 facts["developer"] = developer
@@ -1291,7 +1291,7 @@ def inspect_pkg(input_path, args, facts):
                     if " (" in line:
                         line = line.split(" (")[0]
                     developer = line[len(marker):]
-            if developer != "":
+            if developer not in ("", None):
                 robo_print("Developer is: %s" % developer, LogLevel.VERBOSE, 4)
                 facts["developer"] = developer
             else:
@@ -1340,12 +1340,12 @@ def inspect_pkg(input_path, args, facts):
                     bundle_id = ""
                     if "bundle_id" not in facts:
                         bundle_id = pkginfo_parsed.getroot().attrib["identifier"]
-                    if bundle_id != "":
+                    if bundle_id not in ("", None):
                         robo_print("Bundle identifier: %s" % bundle_id, LogLevel.VERBOSE, 4)
                         facts["bundle_id"] = bundle_id
 
                     install_loc = pkginfo_parsed.getroot().attrib.get("install-location", "")
-                    if install_loc != "":
+                    if install_loc not in ("", None):
                         robo_print("Install location: %s" % install_loc, LogLevel.VERBOSE, 4)
                     else:
                         robo_print("No install location specified", LogLevel.VERBOSE, 4)
@@ -1476,7 +1476,7 @@ def inspect_sourceforge_url(input_path, args, facts):
         proj_name = proj_str[:proj_str.find(marker)]
     else:
         facts["warnings"].append("Unable to parse SourceForge URL.")
-    if proj_name != "":
+    if proj_name not in ("", None):
 
         # Use SourceForge API to obtain project information.
         project_api_url = "https://sourceforge.net/rest/p/" + proj_name
@@ -1518,12 +1518,12 @@ def inspect_sourceforge_url(input_path, args, facts):
         if "app_name" not in facts:
             if "shortname" in parsed_json or "name" in parsed_json:
                 # Record the shortname, if shortname isn't blank.
-                if parsed_json["shortname"] != "":
+                if parsed_json["shortname"] not in ("", None):
                     app_name = parsed_json["shortname"]
                 # Overwrite shortname with name, if name isn't blank.
-                if parsed_json["name"] != "":
+                if parsed_json["name"] not in ("", None):
                     app_name = parsed_json["name"]
-            if app_name != "":
+            if app_name not in ("", None):
                 robo_print("App name is: %s" % app_name, LogLevel.VERBOSE, 4)
                 facts["app_name"] = app_name
 
@@ -1533,7 +1533,7 @@ def inspect_sourceforge_url(input_path, args, facts):
         for this_dict in parsed_json["tools"]:
             if "sourceforge_group_id" in this_dict:
                 proj_id = this_dict["sourceforge_group_id"]
-        if proj_id != "":
+        if proj_id not in ("", None):
             robo_print("SourceForge project ID is: %s" % proj_id, LogLevel.VERBOSE, 4)
             facts["sourceforge_id"] = proj_id
         else:
@@ -1545,11 +1545,11 @@ def inspect_sourceforge_url(input_path, args, facts):
             description = ""
             robo_print("Getting SourceForge description...", LogLevel.VERBOSE)
             if "summary" in parsed_json:
-                if parsed_json["summary"] != "":
+                if parsed_json["summary"] not in ("", None):
                     description = parsed_json["summary"]
-                elif parsed_json["short_description"] != "":
+                elif parsed_json["short_description"] not in ("", None):
                     description = parsed_json["short_description"]
-            if description != "":
+            if description not in ("", None):
                 robo_print("SourceForge description is: %s" % description, LogLevel.VERBOSE, 4)
                 facts["description"] = description
             else:
@@ -1583,7 +1583,7 @@ def inspect_sourceforge_url(input_path, args, facts):
                 if item.find(search).text.startswith("data"):
                     download_url = item.find("link").text.rstrip("/download")
                     break
-            if download_url != "":
+            if download_url not in ("", None):
                 facts = inspect_download_url(download_url, args, facts)
             else:
                 facts["warnings"].append(
@@ -1721,9 +1721,9 @@ def inspect_sparkle_feed_url(input_path, args, facts):
         robo_print("The Sparkle feed does not provide a version "
                    "number", LogLevel.VERBOSE, 4)
     facts["sparkle_provides_version"] = sparkle_provides_version
-    if latest_version != "":
+    if latest_version not in ("", None):
         robo_print("The latest version is %s" % latest_version, LogLevel.VERBOSE, 4)
-    if latest_url != "":
+    if latest_url not in ("", None):
         facts = inspect_download_url(latest_url, args, facts)
 
     # If Sparkle feed is hosted on GitHub or SourceForge, we can gather

